editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,450 @@
+
+void GroupSettingsEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			update_groups();
+		} break;
+	}
+}
+
+void GroupSettingsEditor::_item_edited() {
+	if (updating_group) {
+		return;
+	}
+
+	TreeItem *ti = tree->get_edited();
+	int column = tree->get_edited_column();
+
+	if (!ti) {
+		return;
+	}
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	if (column == 0) {
+		// Name Edited.
+		String new_name = ti->get_text(0);
+		String old_name = ti->get_meta("__name");
+
+		if (new_name == old_name) {
+			return;
+		}
+
+		String error = _check_new_group_name(new_name);
+		if (!error.is_empty()) {
+			ti->set_text(0, old_name);
+			show_message(error);
+			return;
+		}
+
+		String old_property_name = GLOBAL_GROUP_PREFIX + old_name;
+		String new_property_name = GLOBAL_GROUP_PREFIX + new_name;
+
+		String description = ti->get_meta("__description");
+
+		undo_redo->create_action(TTR("Rename Group"));
+
+		undo_redo->add_do_property(ProjectSettings::get_singleton(), new_property_name, description);
+		undo_redo->add_undo_property(ProjectSettings::get_singleton(), new_property_name, Variant());
+
+		undo_redo->add_do_property(ProjectSettings::get_singleton(), old_property_name, Variant());
+		undo_redo->add_undo_property(ProjectSettings::get_singleton(), old_property_name, description);
+
+		undo_redo->add_do_method(this, "rename_references", old_name, new_name);
+		undo_redo->add_undo_method(this, "rename_references", new_name, old_name);
+
+		undo_redo->add_do_method(this, "call_deferred", "update_groups");
+		undo_redo->add_undo_method(this, "call_deferred", "update_groups");
+
+		undo_redo->add_do_method(this, "emit_signal", group_changed);
+		undo_redo->add_undo_method(this, "emit_signal", group_changed);
+
+		undo_redo->commit_action();
+	} else if (column == 1) {
+		// Description Edited.
+		String name = ti->get_text(0);
+		String new_description = ti->get_text(1);
+		String old_description = ti->get_meta("__description");
+
+		if (new_description == old_description) {
+			return;
+		}
+
+		name = GLOBAL_GROUP_PREFIX + name;
+
+		undo_redo->create_action(TTR("Set Group Description"));
+
+		undo_redo->add_do_property(ProjectSettings::get_singleton(), name, new_description);
+		undo_redo->add_undo_property(ProjectSettings::get_singleton(), name, old_description);
+
+		undo_redo->add_do_method(this, "call_deferred", "update_groups");
+		undo_redo->add_undo_method(this, "call_deferred", "update_groups");
+
+		undo_redo->add_do_method(this, "emit_signal", group_changed);
+		undo_redo->add_undo_method(this, "emit_signal", group_changed);
+
+		undo_redo->commit_action();
+	}
+}
+
+void GroupSettingsEditor::_item_button_pressed(Object *p_item, int p_column, int p_id, MouseButton p_button) {
+	if (p_button != MouseButton::LEFT) {
+		return;
+	}
+
+	TreeItem *ti = Object::cast_to<TreeItem>(p_item);
+
+	if (!ti) {
+		return;
+	}
+
+	ti->select(0);
+
+	_show_remove_dialog();
```

Comment:
```suggestion
	ti->select(0);
	_show_remove_dialog();
```
Too many blanks :/
You don't need to separate every single line.